src/Tracing/Span.php

```diff
@@ -15,57 +15,64 @@ class Span implements \JsonSerializable
     /**
      * @var SpanId Span ID
      */
-    private $spanId;
+    protected $spanId;
 
     /**
      * @var TraceId Trace ID
      */
-    private $traceId;
+    protected $traceId;
 
     /**
      * @var string|null Description of the Span
      */
-    private $description;
+    protected $description;
 
     /**
      * @var string|null Operation of the Span
      */
-    private $op;
+    protected $op;
 
     /**
      * @var string|null Completion status of the Span
      */
-    private $status;
+    protected $status;
 
     /**
      * @var SpanId|null ID of the parent Span
      */
-    private $parentSpanId;
+    protected $parentSpanId;
 
     /**
      * @var bool|null Has the sample decision been made?
      */
-    private $sampled;
+    protected $sampled;
 
     /**
      * @var TagsContext A List of tags associated to this Span
      */
-    private $tags;
+    protected $tags;
 
     /**
      * @var Context<mixed> An arbitrary mapping of additional metadata
      */
-    private $data;
+    protected $data;
 
     /**
      * @var float Timestamp in seconds (epoch time) indicating when the span started
      */
-    private $startTimestamp;
+    protected $startTimestamp;
 
     /**
      * @var float|null Timestamp in seconds (epoch time) indicating when the span ended
      */
-    private $endTimestamp;
+    protected $endTimestamp;
+
+    /**
+     * @var SpanRecorder|null Reference instance to the SpanRecorder
+     *
+     * @internal
+     */
+    public $spanRecorder;
```

Comment:
What's the reason of storing the recorder here? We're creating a circular reference between it and the span itself, plus I cannot find any usage of this variable in this class

Reply:
I added implementation now, at least part of it.